multiple_tokens_transfer_and_withdraw/multiple_tokens_transfer_and_withdraw.circom

```diff
@@ -0,0 +1,223 @@
+include "../circomlib/circuits/mimc.circom";
+include "../circomlib/circuits/eddsamimc.circom";
+include "../circomlib/circuits/bitify.circom";
+
+template Main(n,m) {
+
+    // Merkle root of transactions tree
+    signal input tx_root;
+
+    // Merkle proof for transaction in tx tree
+    signal private input paths2tx_root[m-1];
+
+    // binary vector indicating whether node in tx proof is left or right
+    signal private input paths2tx_root_pos[m-1];
+
+    // Merkle root of old balance tree
+    signal input current_state;
+
+    // Merkle proof for sender account in old balance tree
+    signal private input paths2old_root_from[n-1];
+
+    // Merkle proof for receiver account in old balance tree
+    signal private input paths2old_root_to[n-1];
+
+    // Merkle proof for sender account in new balance tree
+    signal private input paths2new_root_from[n-1];
+
+    // Merkle proof for receiver account in new balance tree
+    signal private input paths2new_root_to[n-1];
+
+    // binary vector indicating whether node in balance proof for sender account
+    // is left or right 
+    signal private input paths2root_from_pos[n-1];
+
+    // binary vector indicating whether node in balance proof for receiver account
+    // is left or right 
+    signal private input paths2root_to_pos[n-1];
+    
+    signal private input from_x; //sender address x coordinate
+    signal private input from_y; //sender address y coordinate
+    signal private input R8x; // sender signature
+    signal private input R8y; // sender signature
+    signal private input S; // sender signature
+
+    signal private input nonce_from; // sender account nonce
+    signal private input to_x; // receiver address x coordinate
+    signal private input to_y; // receiver address y coordinate
+    signal private input nonce_to; // receiver account nonce
+    signal private input amount; // amount being transferred
+
+    signal private input token_balance_from; // sender token balance
+    signal private input token_balance_to; // receiver token balance
+    signal private input token_type_from; // sender token type
+    signal private input token_type_to; // receiver token type
+
+    // new balance tree Merkle root
+    signal output out;
+
+    var i;
+
+    var NONCE_MAX_VALUE = 100;
+
+    // constant zero address (EdDSA public key derived from "0x00000..." private key)
```

Comment:
I think this should be the other way around where the public key is 0x00000 and the private key is unknown. That way no one can make transactions that change the balance of this address.